WIP: Added event persistence implementation for StreamsDb

promontis opened the pull request:
@pjvds could you review and add a test

promontis committed:
Added event persistence implementation for StreamsDb (https://streamsdb.io/)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/eventflow/EventFlow?pullRequest=676) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/eventflow/EventFlow?pullRequest=676) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/eventflow/EventFlow?pullRequest=676) it.</sub>

promontis committed:
Added StreamSdb project to sln

promontis committed:
Updated csproj to be in line with other EventFlow projects

rasmus commented:
The tests should as a minimum use [`TestSuiteForEventStore`](https://github.com/eventflow/EventFlow/blob/develop/Source/EventFlow.TestHelpers/Suites/TestSuiteForEventStore.cs) to make sure that the basics are covered.

pjvds commented:
@promontis First of all, thanks for making the effort to write an event store implementation on top of StreamsDB. Very much appreciated!

I've created a pull request that introduces the test project in your fork. This test project is based on the suggestion of @rasmus (thanks for the awesome test set!).

The pull request also has some fixes. If you accept my pull request on your fork, you can test drive the StreamsDB event store in your project.

My pull request updates your develop branch, since EventFlow uses that branch as the trunk. So I would suggest you update this pull request to reflect that change.

See: https://github.com/promontis/EventFlow/pull/1

pjvds commented:
The are still 3 things left to do:

* Support stream deletes (this is blocked by the current beta release of StreamsDB)
* Add readme and example to the EventFlow.EventStore.StreamsDB project
* Add the test project to the CI build

promontis committed:
Merge branch 'develop' into master

promontis committed:
Fixed bug for Int.MaxValue (probably too large)

promontis committed:
Added GroupSubscriber for handling saga's via events from a group stream

promontis committed:
Added StreamsDbReadModelStore which writes/reads readmodels using streams

promontis committed:
Merge branch 'master' of https://github.com/promontis/EventFlow

promontis committed:
Added a ReadStoreManager based on subscription instead of in-process forwards.

promontis committed:
Injecting SubscriptionBasedReadStoreManager
Added CursorsStream to ReadModuleUpdate poco
Working test, but still need to add transaction when psa is done

promontis committed:
Syncing work from Stylister repository

rasmus committed:
Merge branch 'develop-v0' into master
